tests/e2e/healthcheck_test.go

```
@@ -0,0 +1,68 @@
+	defer cancel()
+	if err := redisPing(ctx, rawURL); err != nil {
+		t.Fatalf("E2E Redis not reachable: %v", err)
+	}
+	t.Logf("E2E Redis healthy at %s", rawURL)
+}
+
+// TestHealthCheck_ServerHTTP verifies the E2E server responds to /healthz.
+func TestHealthCheck_ServerHTTP(t *testing.T) {
+	baseURL := e2eServerURL(t)
+	client := &http.Client{Timeout: 3 * time.Second}
+	resp, err := client.Get(baseURL + "/healthz")
```

golangci-lint (2.12.2)

[error] (*net/http.Client).Get must not be called. use (*net/http.Client).Do(*http.Request)

(noctx)